### ISSUE TITLE
Unable to process webmention from mastodon

### ISSUE THREAD

**Issue by RobbiNespu** [aaronpk/webmention.io#184]
source:  https://mastodon.radio/@robbinespu/109619770124693455
target: https://journal.robbi.my/notes/230102050755/

```
 $ curl -i -d source=https://mastodon.radio/@robbinespu/109619770124693455 -d target=https://journal.robbi.my/notes/230102050755/ https://webmention.io/journal.robbi.my/webmention
HTTP/2 201
content-type: application/json;charset=UTF-8
content-length: 273
location: https://webmention.io/journal.robbi.my/webmention/Buyy2WmpSgFf5_dFyKFh
status: 201 Created
cache-control: no-store
access-control-allow-origin: *
x-content-type-options: nosniff
date: Mon, 02 Jan 2023 20:19:53 GMT
x-powered-by: Phusion Passenger 5.3.1
server: nginx/1.14.0 + Phusion Passenger 5.3.1

{"status":"queued","summary":"Webmention was queued for processing","location":"https://webmention.io/journal.robbi.my/webmention/Buyy2WmpSgFf5_dFyKFh","source":"https://mastodon.radio/@robbinespu/109619770124693455","target":"https://journal.robbi.my/notes/230102050755/"}

```

from web panel, https://webmention.io/journal.robbi.my/webmention/Buyy2WmpSgFf5_dFyKFh

Debug Info
```
{
  "status": "no_link_found",
  "source": "https://mastodon.radio/@robbinespu/109619770124693455",
  "target": "https://journal.robbi.my/notes/230102050755/",
  "summary": "The source document does not have a link to the target URL"
}
```

Checking the parse https://xray.p3k.io/parse?url=https%3A%2F%2Fmastodon.radio%2F%40robbinespu%2F109619770124693455&pretty=true

```json
{
    "data": {
        "type": "entry",
        "url": "https://mastodon.radio/@robbinespu/109619770124693455",
        "published": "2023-01-02T13:09:09+00:00",
        "content": {
            "text": "After 15 years, I finally took an exam for amateur radio license https://journal.robbi.my/notes/230102050755/Horayy \ud83d\ude01",
            "html": "<p>After 15 years, I finally took an exam for amateur radio license </p><p><a href=\"https://journal.robbi.my/notes/230102050755/\"><span>https://</span><span>journal.robbi.my/notes/2301020</span><span>50755/</span></a></p><p>Horayy \ud83d\ude01</p>"
        },
        "author": {
            "type": "card",
            "name": "robbinespu",
            "url": "https://mastodon.radio/@robbinespu",
            "photo": "https://mastodon.radio/system/accounts/avatars/000/090/792/original/cca95e6eb051896a.png",
            "nickname": "robbinespu"
        },
        "post-type": "note"
    },
    "url": "https://mastodon.radio/@robbinespu/109619770124693455",
    "code": 200,
    "source-format": "activity+json"
}

```

That all I have, not sure why I getting `no_link_found`

**Comment by aaronpk** [comment]
Unfortunately there isn't a good solution for this. Mastodon stopped returning HTML and it's now a big JS app. So it's true that there is in fact no link from the mastodon post URL to your target URL.

The XRay parse you found does some trickery with Accept headers to fetch Mastodon's ActivityPub JSON and translates it to Microformats. However, webmention.io tells XRay to prioritize fetching HTML content, which results in Mastodon returning the empty HTML app shell.

In other words, I don't think this is going to be possible to support now that Mastodon went js;dr
